core/src/main/java/discord4j/core/shard/GatewayBootstrap.java

```diff
@@ -774,11 +773,11 @@ public StoreService adaptStoreService(StoreService storeService) {
                             .checkpoint("Read payload from gateway")
                             .flatMap(dispatch -> {
                                 ShardInfo info;
-                                Dispatch actual;
-                                if (dispatch instanceof ShardAwareDispatch) {
-                                    ShardAwareDispatch shardDispatch = (ShardAwareDispatch) dispatch;
+                                LazyDispatch<?> actual;
+                                if (dispatch.isShardAware()) {
+                                    ShardAwareDispatch shardDispatch = (ShardAwareDispatch) dispatch.getData();
                                     info = new ShardInfo(shardDispatch.getShardIndex(), shardDispatch.getShardCount());
```

Comment:
Can we avoid the nullable IDE warning here?

Reply:
It is implicitly checked PayloadHandlers.java, but the field is marked nullable and therefore these warnings happen around everywhere we use this class.
I'll have a look if I can remove the "nullable" annotation.